### PR TITLE
updates

### PR DESCRIPTION
**Pull request by AveryGC**
updated project

### CODE REVIEW

File: LMS_Admin/src/main/java/com/SS/library/Controller/AuthorController.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			return new ResponseEntity<List<Author>>(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+	}
+	
+	@RequestMapping(path = "**/Admin/authors", method = {RequestMethod.POST, RequestMethod.DELETE, RequestMethod.PUT})
+	public ResponseEntity<String> authorMethodNotAllowed() {
+		return new ResponseEntity<String>(HttpStatus.METHOD_NOT_ALLOWED);
+	}
```
Comment (James-H2o) [inline LMS_Admin/src/main/java/com/SS/library/Controller/AuthorController.java:48]: Why did you add this?

File: LMS_Admin/src/main/java/com/SS/library/Controller/GenreController.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		}
+	}
+	@RequestMapping(path = "**/Admin/genres/{genreId}", method = RequestMethod.POST)
+	public ResponseEntity<Genre> addgenre(@RequestBody Genre genre){
+		try {
+			genreService.addGenre(genre);
+			return new ResponseEntity<Genre>(genre,HttpStatus.ACCEPTED);
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			return new ResponseEntity<Genre>(HttpStatus.CONFLICT);
```
Comment (James-H2o) [inline LMS_Admin/src/main/java/com/SS/library/Controller/GenreController.java:58]: What was your reasoning on your choice of HttpStatus responses?

File: LMS_Admin/src/main/java/com/SS/library/DAO/DAO.java [hunk 3]
```diff
@@ -33,7 +30,7 @@ protected void save(String sql, Object[] vals) throws SQLException {
     }
 
     protected Integer saveRecieveKey(String sql, Object[] vals) throws SQLException {
```
Comment (James-H2o) [inline LMS_Admin/src/main/java/com/SS/library/DAO/DAO.java:32]: Why two pstmt?

File: LMS_Admin/src/main/java/com/SS/library/Utility/ConnectUtil.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
     public ConnectUtil() throws SQLException, ClassNotFoundException {
         Class.forName(driver);
         conn = DriverManager.getConnection(url,username,password);
-        conn.setAutoCommit(false);
+//        conn.setAutoCommit(false);
```
Comment (James-H2o) [inline LMS_Admin/src/main/java/com/SS/library/Utility/ConnectUtil.java:21]: Pramod's example has this as a protected method. What would be the pros and cons?

File: LMS_Admin/src/test/java/com/SS/library/DAO/AuthorDAOTest.java [hunk 5]
```diff
@@ -31,8 +31,8 @@
 //	ConnectUtil connectUtil;
 //	@Mock
 //	private AuthorDAO authorDAO;
-//
-//
+////
+////
 //	@Test
 //	void testRead() {
```
Comment (James-H2o) [inline LMS_Admin/src/test/java/com/SS/library/DAO/AuthorDAOTest.java:37]: The code is commented out. Did you end up implementing this?

**Review by James-H2o — COMMENTED** [review]
Looks good overall. Most of my comments are conversational, something to think about. Obviously the commented out code needs to be addressed.

**Review by James-H2o — COMMENTED** [review]
Had to view one more file

**Review by srv-max — APPROVED** [review]
